src/webgpu/shader/validation/shader_validation_test.ts

```diff
@@ -12,12 +12,10 @@ export class ShaderValidationTest extends GPUTest {
    * ```ts
    * t.expectCompileResult(true, `wgsl code`); // Expect success
    * t.expectCompileResult(false, `wgsl code`); // Expect validation error with any error string
-   * t.expectCompileResult('substr', `wgsl code`); // Expect validation error containing 'substr'
    * ```
-   *
-   * TODO(gpuweb/gpuweb#1813): Remove the "string" overload if there are no standard error codes.
+   * foobar
    */
-  expectCompileResult(expectedResult: boolean | string, code: string) {
+  expectCompileResult(expectedResult: boolean, code: string) {
```

Comment:
I would suggest to  keep this as the string expectedResult can handy when want to check a test is failing with an expected error message (instead of a specific error code)?

Unless failures for wrong reasons is not a real concern or expecting a specific substring in the error message introduces the same problems as we had with error codes.

Reply:
I think the WGSL spec doesn't specify particular strings that must appear in messages for particular errors, so we can't test them. Unless we added some way for each browser to customize the strings they expect for every test.

Without a way to test which error occurred, we have to instead test that a new error arises from a minimal code change between a control case and a test case.